src/Features/LanguageServer/Protocol/CustomProtocol/FindUsagesLSPContext.cs

```diff
@@ -133,15 +159,25 @@ public override async Task OnReferenceFoundAsync(SourceReferenceItem reference)
 
                 if (referenceItem != null)
                 {
-                    AddToReferencesToReport_MustBeCalledUnderLock(referenceItem);
+                    AddToReferencesToReport_MustBeCalledUnderLock(referencesToReport, referenceItem);
                 }
             }
+
+            ReportIfNotEmpty(referencesToReport);
         }
 
-        private void AddToReferencesToReport_MustBeCalledUnderLock(VSReferenceItem item)
+        private void AddToReferencesToReport_MustBeCalledUnderLock(ArrayBuilder<VSReferenceItem> referencesToReport, VSReferenceItem item)
         {
             Debug.Assert(_semaphore.CurrentCount == 0);
+
             _resultsChunk.Add(item);
+            if (_resultsChunk.Count < MaxResultsChunkSize)
```

Comment:
i don't quite get the purpose of hte chunking.  is it necessary?

Reply:
I believe the chunking can help improve performance, especially when there are a lot of references. I remember previously testing with finding references for SyntaxKind and chunking was much faster than individually reporting references.

Reply:
do you know who owns the IProgress impl?  If chunking helps, it feels like that shoud happen at their layer.  otherwise all clients have to write a chunking system.